### PR TITLE
Integrated signup FE/BE

### PR DESCRIPTION
**Pull request by akhilarjun3-sureify**
Screenshot from 2023-06-05 12-26-19

### CODE REVIEW

**Review by maniknr-sureify — APPROVED** [review]

**Review by harshavardhan-sureify — APPROVED** [review]

File: src/components/SignupPage.jsx [hunk 1]
```diff
@@ -37,18 +41,44 @@ const SignupPage = () => {
...
       if (res.status === 200) {
+        const data = res.data.data
+         
+        localStorage.setItem("userToken",  data.token);
+        localStorage.setItem("role", data.role);
+        navigate("/dashborad");
       }
-    } catch (err) {}
+    } catch (err) {
+      const response = err.response;
```
Comment (Madhukar-Sureify) [inline src/components/SignupPage.jsx:62]: add toasters at right top 
image

Comment (akhilarjun3-sureify) [reply src/components/SignupPage.jsx:62]: changed.

File: src/components/SignupPage.jsx [hunk 2]
```diff
@@ -20,7 +24,7 @@ import logo from "../assets/images/logo.png";
 import profileImage from "../assets/images/profileImage.png";
 import { ImagePaper } from "./Styles";
 import { ImageText } from "./Styles";
```
Comment (abhishekdundigalla-sureify) [inline src/components/SignupPage.jsx:26]: you can write above two lines in a single line

File: src/components/SignupPage.jsx [hunk 3]
```diff
@@ -37,18 +41,44 @@ const SignupPage = () => {
...
+      const postData = { ...signUpdata }
+      delete postData.cpassword
+      const res = await axios.post(signUp, postData);
+      
       if (res.status === 200) {
+        const data = res.data.data
+         
+        localStorage.setItem("userToken",  data.token);
+        localStorage.setItem("role", data.role);
+        navigate("/dashborad");
```
Comment (abhishekdundigalla-sureify) [inline src/components/SignupPage.jsx:59]: check the spelling of dashboard

File: src/components/SignupPage.jsx [hunk 4]
```diff
@@ -37,18 +41,44 @@ const SignupPage = () => {
...
       }
-    } catch (err) {}
+    } catch (err) {
+      const response = err.response;
+
+      if (response) {
```
Comment (abhishekdundigalla-sureify) [inline src/components/SignupPage.jsx:64]: why to take another response variable? In if condition you can directly check err.response

File: src/components/SignupPage.jsx [hunk 5]
```diff
@@ -86,7 +117,9 @@ const SignupPage = () => {
         if (!value) {
           message = "Name is required";
         } else if (value.length < 3) {
-          message = "Name should have atleast length 3";
+          message = "Name should have atleast 3 characters";
+        } else if (!/^[A-Za-z\s]+$/.test(value)) {
+          message = "Name should not have digits";
```
Comment (abhishekdundigalla-sureify) [inline src/components/SignupPage.jsx:122]: change "have" to "contain"

File: src/components/SignupPage.jsx (hunk 3 above)
Comment (akhilarjun3-sureify) [reply src/components/SignupPage.jsx]: changed

File: src/components/SignupPage.jsx (hunk 5 above)
Comment (akhilarjun3-sureify) [reply src/components/SignupPage.jsx:119]: changed

File: src/components/SignupPage.jsx (hunk 2 above)
Comment (akhilarjun3-sureify) [reply src/components/SignupPage.jsx]: changed

File: src/components/SignupPage.jsx (hunk 4 above)
Comment (akhilarjun3-sureify) [reply src/components/SignupPage.jsx]: modified it

**Review by abhishekdundigalla-sureify — APPROVED** [review]

**Review by Madhukar-Sureify — APPROVED** [review]

**Pull request merged**